iothub/device/src/InternalClient.cs

```diff
@@ -57,58 +50,46 @@ internal class InternalClient : IDisposable
 
         private int _diagnosticSamplingPercentage;
 
-        protected internal InternalClient(
+        internal InternalClient(
             IotHubConnectionCredentials iotHubConnectionCredentials,
-            IotHubClientOptions iotHubClientOptions,
-            IDeviceClientPipelineBuilder pipelineBuilder)
+            IotHubClientOptions iotHubClientOptions)
         {
-            Argument.AssertNotNull(iotHubClientOptions, nameof(iotHubClientOptions));
-
             if (Logging.IsEnabled)
-                Logging.Enter(this, iotHubClientOptions.TransportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
+                Logging.Enter(this, iotHubClientOptions?.TransportSettings, nameof(InternalClient) + "_ctor");
+
+            // Make sure client options is initialized.
+            if (iotHubClientOptions == default)
+            {
+                iotHubClientOptions = new();
+            }
 
             IotHubConnectionCredentials = iotHubConnectionCredentials;
             _clientOptions = iotHubClientOptions;
 
-            if (!iotHubClientOptions.ModelId.IsNullOrWhiteSpace()
-                && iotHubClientOptions.TransportSettings is IotHubClientHttpSettings)
+            if (!_clientOptions.ModelId.IsNullOrWhiteSpace()
+                && _clientOptions.TransportSettings is IotHubClientHttpSettings)
```

Comment:
Aren't we already not allowing users to use HTTP? We can get rid of this check, if so

Reply:
good point